### PR TITLE
remove repository fallback

### PR DESCRIPTION
**Pull request by Sertonix**
This removes the a fallback url for core packages that are missing the `repository` parameter in the `package.json`.
This fallback value is not correct for packages in the main repo and for packages where the name differes from the repo name like [`language-rust-bundled`](https://github.com/pulsar-edit/language-rust).
Also the `repository` is and should always be given for core packages.

This is a bit conroversial to this pull rquest: #253

### CODE REVIEW

File: packages/settings-view/lib/installed-packages-panel.js [hunk 1]
```diff
@@ -148,12 +148,6 @@ export default class InstalledPackagesPanel extends CollapsibleSectionPanel {
     packages.core = packages.core.filter(({theme}) => !theme)
     packages.git = (packages.git || []).filter(({theme}) => !theme)
 
-    for (let pack of packages.core) {
```
Comment (mauricioszabo) [inline packages/settings-view/lib/installed-packages-panel.js:151]: Just a test

**Review by mauricioszabo — CHANGES_REQUESTED** [review]
Ignore this, it's just a test...

File: packages/settings-view/lib/installed-packages-panel.js (hunk 1 above)
Comment (Sertonix) [reply packages/settings-view/lib/installed-packages-panel.js:151]: Can I resolve this?

**Review by confused-Techie — APPROVED** [review]
I know this one has been around for a while (and we are having an unrelated action failure)

But on further inspection I fully support the work of this PR, and think we should go ahead and merge it.

The only icing on the cake to this PR would be to ensure that all core packages properly declare a repository, as I assume the lack of doing so is what caused this feature in the first place. But regardless, we should probably implement this, as it's logic is likely not very helpful nowadays.

**Comment by Sertonix** [comment]
Maybe this could be verified at compile time or in a test. This would make sure that this will not occur again.

**Comment by confused-Techie** [comment]
> Maybe this could be verified at compile time or in a test. This would make sure that this will not occur again.

It could, although now since nearly every single bundled package is available right within this one repo, it should be pretty easy at a glance to confirm that everything has the proper repository. Which when I get a moment I'll go ahead and verify

**Comment by confused-Techie** [comment]
So I wanted to quickly confirm that this logic is no longer needed. To make life super simple, I put together a quick script that will grab the repository field from all core bundled packages:

<details>
<summary>Script</summary>

```javascript
const fs = require("fs");
const path = require("path");

const dirs = fs.readdirSync("./packages");

let out = [];

for (const dir of dirs) {
  let packPath = `./packages/${dir}/package.json`;

  if (!fs.lstatSync(`./packages/${dir}`).isDirectory()) {
    continue;
  }
  
  let packJson = fs.readFileSync(packPath, { encoding: "utf8" });
  packJson = JSON.parse(packJson);

  out.push(`Package: ${dir} - Repo: ${packJson.repository}`);
}

console.log(out.join("\n"));
```

</details>

Which gives us the following data:

<details>
<summary>Data</summary>

```
Package: about - Repo: https://github.com/pulsar-edit/pulsar
Package: archive-view - Repo: https://github.com/pulsar-edit/pulsar
Package: atom-dark-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: atom-dark-ui - Repo: https://github.com/pulsar-edit/pulsar
Package: atom-light-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: atom-light-ui - Repo: https://github.com/pulsar-edit/pulsar
Package: autocomplete-atom-api - Repo: https://github.com/pulsar-edit/pulsar
Package: autocomplete-css - Repo: https://github.com/pulsar-edit/pulsar
Package: autocomplete-html - Repo: https://github.com/pulsar-edit/pulsar
Package: autocomplete-plus - Repo: https://github.com/pulsar-edit/pulsar
Package: autocomplete-snippets - Repo: https://github.com/pulsar-edit/pulsar
Package: autoflow - Repo: https://github.com/pulsar-edit/pulsar
Package: autosave - Repo: https://github.com/pulsar-edit/pulsar
Package: background-tips - Repo: https://github.com/pulsar-edit/pulsar
Package: base16-tomorrow-dark-theme - Repo: https://github.com/pulsar-edit/pulsar
Package: base16-tomorrow-light-theme - Repo: https://github.com/pulsar-edit/pulsar
Package: bookmarks - Repo: https://github.com/pulsar-edit/pulsar
Package: bracket-matcher - Repo: https://github.com/pulsar-edit/pulsar
Package: command-palette - Repo: https://github.com/pulsar-edit/pulsar
Package: dalek - Repo: https://github.com/pulsar-edit/pulsar
Package: deprecation-cop - Repo: https://github.com/pulsar-edit/pulsar
Package: dev-live-reload - Repo: https://github.com/pulsar-edit/pulsar
Package: encoding-selector - Repo: https://github.com/pulsar-edit/pulsar
Package: exception-reporting - Repo: https://github.com/pulsar-edit/pulsar
Package: find-and-replace - Repo: https://github.com/pulsar-edit/find-and-replace
Package: fuzzy-finder - Repo: https://github.com/pulsar-edit/fuzzy-finder
Package: git-diff - Repo: https://github.com/pulsar-edit/pulsar
Package: go-to-line - Repo: https://github.com/pulsar-edit/pulsar
Package: grammar-selector - Repo: https://github.com/pulsar-edit/pulsar
Package: image-view - Repo: https://github.com/pulsar-edit/pulsar
Package: incompatible-packages - Repo: https://github.com/pulsar-edit/pulsar
Package: keybinding-resolver - Repo: https://github.com/pulsar-edit/pulsar
Package: language-c - Repo: https://github.com/pulsar-edit/pulsar
Package: language-clojure - Repo: https://github.com/pulsar-edit/pulsar
Package: language-coffee-script - Repo: https://github.com/pulsar-edit/pulsar
Package: language-csharp - Repo: https://github.com/pulsar-edit/pulsar
Package: language-css - Repo: https://github.com/pulsar-edit/pulsar
Package: language-gfm - Repo: https://github.com/pulsar-edit/pulsar
Package: language-git - Repo: https://github.com/pulsar-edit/pulsar
Package: language-go - Repo: https://github.com/pulsar-edit/pulsar
Package: language-html - Repo: https://github.com/pulsar-edit/pulsar
Package: language-hyperlink - Repo: https://github.com/pulsar-edit/pulsar
Package: language-java - Repo: https://github.com/pulsar-edit/pulsar
Package: language-javascript - Repo: https://github.com/pulsar-edit/pulsar
Package: language-json - Repo: https://github.com/pulsar-edit/pulsar
Package: language-less - Repo: https://github.com/pulsar-edit/pulsar
Package: language-make - Repo: https://github.com/pulsar-edit/pulsar
Package: language-mustache - Repo: https://github.com/pulsar-edit/pulsar
Package: language-objective-c - Repo: https://github.com/pulsar-edit/pulsar
Package: language-perl - Repo: https://github.com/pulsar-edit/pulsar
Package: language-php - Repo: https://github.com/pulsar-edit/pulsar
Package: language-property-list - Repo: https://github.com/pulsar-edit/pulsar
Package: language-python - Repo: https://github.com/pulsar-edit/pulsar
Package: language-ruby - Repo: https://github.com/pulsar-edit/pulsar
Package: language-ruby-on-rails - Repo: https://github.com/pulsar-edit/pulsar
Package: language-rust-bundled - Repo: https://github.com/pulsar-edit/pulsar
Package: language-sass - Repo: https://github.com/pulsar-edit/pulsar
Package: language-shellscript - Repo: https://github.com/pulsar-edit/pulsar
Package: language-source - Repo: https://github.com/pulsar-edit/pulsar
Package: language-sql - Repo: https://github.com/pulsar-edit/pulsar
Package: language-text - Repo: https://github.com/pulsar-edit/pulsar
Package: language-todo - Repo: https://github.com/pulsar-edit/pulsar
Package: language-toml - Repo: https://github.com/pulsar-edit/pulsar
Package: language-typescript - Repo: https://github.com/pulsar-edit/pulsar
Package: language-xml - Repo: https://github.com/pulsar-edit/pulsar
Package: language-yaml - Repo: https://github.com/pulsar-edit/pulsar
Package: line-ending-selector - Repo: https://github.com/pulsar-edit/pulsar
Package: link - Repo: https://github.com/pulsar-edit/pulsar
Package: markdown-preview - Repo: https://github.com/pulsar-edit/pulsar
Package: notifications - Repo: https://github.com/pulsar-edit/pulsar
Package: one-dark-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: one-dark-ui - Repo: https://github.com/pulsar-edit/pulsar
Package: one-light-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: one-light-ui - Repo: https://github.com/pulsar-edit/pulsar
Package: open-on-github - Repo: https://github.com/pulsar-edit/pulsar
Package: package-generator - Repo: https://github.com/pulsar-edit/pulsar
Package: pulsar-updater - Repo: https://github.com/pulsar-edit/pulsar
Package: settings-view - Repo: https://github.com/pulsar-edit/pulsar
Package: solarized-dark-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: solarized-light-syntax - Repo: https://github.com/pulsar-edit/pulsar
Package: spell-check - Repo: https://github.com/pulsar-edit/pulsar
Package: status-bar - Repo: https://github.com/pulsar-edit/pulsar
Package: styleguide - Repo: https://github.com/pulsar-edit/pulsar
Package: tabs - Repo: https://github.com/pulsar-edit/pulsar
Package: timecop - Repo: https://github.com/pulsar-edit/pulsar
Package: tree-view - Repo: https://github.com/pulsar-edit/pulsar
Package: update-package-dependencies - Repo: https://github.com/pulsar-edit/pulsar
Package: welcome - Repo: https://github.com/pulsar-edit/pulsar
Package: whitespace - Repo: https://github.com/pulsar-edit/pulsar
Package: wrap-guide - Repo: https://github.com/pulsar-edit/pulsar
```

</details>

Which using that, we can confirm:

* Every single core bundled package properly declares it's repository
* Nearly every single one has Pulsar as it's repo (as it should) except: `find-and-replace` and `fuzzy-finder` (Which these probably should be fixed at some point)

With all of that said, looks like we are good to remove this behavior, assuming we ensure to keep our packages proper in the future, which seems very likely that we will.

And on that not, I think I'll give this a little time in case anyone has a response on this new discussion, or otherwise I'll merge these changes.

As always, and thanks again @Sertonix for the fantastic contribution, and your extreme patience

**Comment by confused-Techie** [comment]
Alright, with no other comments made in a short amount of time, factoring in that this PR has been open for months, I'll go ahead and merge it.

The one build failure we are seeing here, is due to some issues that have just been resolved over on #698

**Comment by DeeDeeG** [comment]
I think this was a data robustness thing.

Perhaps returning some string or even some URL is better, even if it's the trying-to-be-too-clever URL they were constructing before -- rather than potentially allowing it to return `null`.

Maybe something obviously wrong like `https://eample.com/MISSING_REPO_UPDATE_PACKAGEJSON`, so it will get attention to be fixed with a proper URL.

(Sorry for not responding earlier. And this is for hypothetically we bundle another package but miss it not having a `repository` field, which I guess may never happen... I kinda want to return some string or URL, though.)

**Comment by confused-Techie** [comment]
> I think this was a "don't trust your inputs" data safety thing.
> 
> Perhaps returning some string or even some URL is better, even if it's _not_ the trying-to-be-too-clever URL they were constructing before -- rather than potentially allowing it to return `null`. I dunno if stuff breaks if we return `null` for the `pack.repository` here.
> 
> Maybe something obviously wrong like `https://example.com/MISSING_REPOSITORY_PLEASE_UPDATE_PACKAGEJSON`, so someone will notice, and it will get attention to be fixed with a proper URL, but without maybe breaking the app.
> 
> (Sorry for not responding earlier. And this is for hypothetically we bundle another package but miss it not having a `repository` field, which I guess may never happen... I kinda want to return some string or URL, though.)

No sorry needed for not responded earlier.
That two hour wait was more of a mull it over, since I'd assumed with this PR having been around for so long, most people were indifferent about it's behavior. Assuming then it'd be safe to remove. 

I'm not aware of anything that would break without this later on, although a good point. Since honestly with missing data, it feels more right that we would check more fields that I'd assume cause more breakage, like `name` or `version`. But can see how maybe this was implemented because of a particular issue.

I really just felt uneasy about such an old PR not getting the attention deserved, and wanted to get it in, but if you feel it's better to return something, I'd have no concerns about reverting this, or reimplementing some sort of logic to help institute those protections. 

The tests at least seemed happy, but in the future maybe there could be some sort of issue

**Comment by DeeDeeG** [comment]
Okay, after some research and looking at it more closely, this seems not so high-stakes after all and fine just the way it was done here.

On closer look, this code is more of a "leaf" or edge consumer type of code, not a bit of trunk or a really central branch thing that everything else relies on.

Also, the original justification was apparently just to get the correct Atom org icon applied to core packages.

- Original commit adding this: https://github.com/pulsar-edit/settings-view/commit/3b74b350a10d0cc59786003a01376ac259cc56f8
- So, if the fix was implemented for cosmetic purposes, then the harms for not normalizing this data can't be too bad. (I mean, it's within reason and a tolerable risk, or something like that. To be less over-optimistic than saying "can't be too bad", famous last words.)
- If the expected consequence is that the org owner icon breaks for packages without a `repository` fiel filled out properly in their package.json files, then I suppose anyone who reads this can **be on the lookout for core packages whose org owner icon is blank or bugged out some way, etc.**

Not too bad all around, in which case, yeah I am supportive of the removal and it doesn't need some other way of normalizing the data.

**Comment by DeeDeeG** [comment]
Looking at that initial commit (https://github.com/pulsar-edit/settings-view/commit/3b74b350a10d0cc59786003a01376ac259cc56f8) that implemented this `repository` value guessing code, it's in another file `settings-view/lib/themes-panel.js`.

So, for consistency it should be done there, too. For built-in themes, as well not just built-in non-theme "Packages", since the "Themes" panel is separate from the "Packages" panel, I suppose.

https://github.com/pulsar-edit/pulsar/blob/4592d1849902e2814caa895ca96bb3b84afbb2fa/packages/settings-view/lib/themes-panel.js#L203-L207

I can do a quick follow-up PR to make it consistent with this one.

**Comment by confused-Techie** [comment]
@DeeDeeG As always love how thorough your work is.

You seem to be totally correct that this was a cosmetic fix, and isn't something we have to worry to much about, and good find on it's sibling behavior elsewhere in the codebase.

I'd happily and quickly approve any PRs addressing that, as I can see you've already made the changes needed for it.

And with this moved, who knows, it'll be nice getting us 0.00001 ms faster startup (or something lol). But glad we can confirm there's no big deal breakers in this one

**Comment by DeeDeeG** [comment]
@confused-Techie okay, I did it, see: #702 👍

**Pull request merged**